### DISCUSSION TITLE
MySQL (Amazon RDS) connections "expire" silently; further queries hang DBeaver

### DISCUSSION THREAD

**Discussion by JonathanHarford** [dbeaver/dbeaver#287] [Q&A] [answered]
When I connect to any of my Amazon RDS databases, wait a while (~20 minutes) and then run _any_ query, DBeaver behaves as though the query is taking a long time to run.

If I click "Cancel" the button simply turns grey. Whether I hit cancel or not, I am eventually told:

> DBCException: SQL Error [08S01]: Communications link failure
> The last packet successfully received from the server was 2,262,238 milliseconds ago.  The last packet sent successfully to the server was 955,700 milliseconds ago.
> SQL Error [08S01]: Communications link failure
> The last packet successfully received from the server was 2,262,238 milliseconds ago.  The last packet sent successfully to the server was 955,700 milliseconds ago.
>   CommunicationsException: Communications link failure
> The last packet successfully received from the server was 2,262,238 milliseconds ago.  The last packet sent successfully to the server was 955,700 milliseconds ago.
>   Communications link failure
> The last packet successfully received from the server was 2,262,238 milliseconds ago.  The last packet sent successfully to the server was 955,700 milliseconds ago.
>     SocketException: Connection timed out
>     Connection timed out

This is on DBeaver 3.6.2 (and many previous) for Ubuntu x64.

**Comment by codebling** [comment]
Once you eventually get the "Communications link failure", DBeaver is responsive again, right?  It's simply while it's trying to run the query that DBeaver is unresponsive?  How long does this last? 

This will be fixed by #6.

**Comment by JonathanHarford** [comment]
Yes, DBeaver is responsive again after Communications link failure. Attempting to run the query again at that point gets:

```
JDBCException: SQL Error [08003]: No operations allowed after connection closed.
```

When I suspect a connection's been open too long, I Disconnect and then Connect, Almost always I can then query on that connection -- but sometimes (like 10% of the time) the disconnecting will cause it to hang.

And I realize that it's only _that_ connection that becomes unresponsive. I can select and query other connections while waiting for Communications link failure.

How long does this last? 1095 seconds, at last attempt.

Great to hear a fix is in the works!

**Comment by codebling** [comment]
So the _unresponsiveness_ lasted ~18 minutes??   Which driver are you using?

**Comment by JonathanHarford** [comment]
Yeah, 18 minutes of unresponsiveness. Usually I kill and restart DBeaver, 

It's the "MySQL standard driver".

**Comment by serge-rider** [comment, marked as answer]
Most likely server kills connection after some period of time.
Workaround: you can set driver property socketTimeout to some reasonable value (e.g. 20 seconds).
There will be one nasty side-effect. Long-running queries will fail if they will execute longer than socketTimeout.
Another workaround is to turn on connection keep-alive. This feature will be added in 3.6.3 (#6)

**Comment by JonathanHarford** [comment]
Thank you so much!
